src/web/services/feedback-responses.service.spec.ts

```diff
@@ -1,16 +1,531 @@
+      get: jest.fn(),
+      post: jest.fn(),
+      put: jest.fn(),
+      delete: jest.fn(),
+    };
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+      ],
+      providers: [
+        { provide: HttpRequestService, useValue: spyHttpRequestService },
+      ],
+    });
+    service = TestBed.inject(FeedbackResponsesService);
+  });
 
   it('should be created', () => {
-    const service: FeedbackResponsesService = TestBed.inject(FeedbackResponsesService);
     expect(service).toBeTruthy();
   });
+
+  it('should retrieve the correct default essay/text response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.TEXT))
+      .toStrictEqual(DEFAULT_TEXT_RESPONSE_DETAILS());
+  });
+
+  it('should retrieve the correct default rank (options) response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.RANK_OPTIONS))
+      .toStrictEqual(DEFAULT_RANK_OPTIONS_RESPONSE_DETAILS());
+  });
+
+  it('should retrieve the correct default rank (recipients) response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.RANK_RECIPIENTS))
+      .toStrictEqual(DEFAULT_RANK_RECIPIENTS_RESPONSE_DETAILS());
+  });
+
+  it('should retrieve the correct default team contribution response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.CONTRIB))
+      .toStrictEqual(DEFAULT_CONTRIBUTION_RESPONSE_DETAILS());
+  });
+
+  it('should retrieve the correct default numerical scale response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.NUMSCALE))
+      .toStrictEqual(DEFAULT_NUMSCALE_RESPONSE_DETAILS());
+  });
+
+  it('should retrieve the correct default MCQ response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.MCQ))
+      .toStrictEqual(DEFAULT_MCQ_RESPONSE_DETAILS());
+  });
+
+  it('should retrieve the correct default MSQ response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.MSQ))
+      .toStrictEqual(DEFAULT_MSQ_RESPONSE_DETAILS());
+  });
+
+  it('should retrieve the correct default rubric response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.RUBRIC))
+      .toStrictEqual(DEFAULT_RUBRIC_RESPONSE_DETAILS());
+  });
+
+  it('should retrieve the correct default distribute points/constant sum (options) response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.CONSTSUM_OPTIONS))
+      .toStrictEqual(DEFAULT_CONSTSUM_RESPONSE_DETAILS());
+  });
+
+  it('should retrieve the correct default distribute points/constant sum (recipients) response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.CONSTSUM_RECIPIENTS))
+      .toStrictEqual(DEFAULT_CONSTSUM_RESPONSE_DETAILS());
+  });
```

Comment:
Only the parameters seem to be different across all these test cases. Why not use a `Record`/List to store pairs of `QuestionType` and Response details and then loop through them? Or add them in the same test case?